src/libs/actions/Transaction.ts

```diff
@@ -264,4 +271,52 @@ function clearError(transactionID: string) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`, {errors: null});
 }
 
-export {addStop, createInitialWaypoints, saveWaypoint, removeWaypoint, getRoute, updateWaypoints, clearError};
+function markAsCash(transactionID: string, transactionThreadReportID: string) {
+    const optimisticReportAction = buildOptimisticDismissedViolationReportAction({
+        reason: 'manual',
+        violationName: CONST.VIOLATIONS.RTER,
+    });
+    const optimisticReportActions = {
+        [optimisticReportAction.reportActionID]: optimisticReportAction,
+    };
+    const onyxData: OnyxData = {
+        optimisticData: [
+            // Optimistically dismissing the violation, removing it from the list of violations
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`,
+                value: allTransactionViolations.filter((violation: TransactionViolation) => violation.name !== CONST.VIOLATIONS.RTER),
+            },
+            // Optimistically adding the system message indicating we dismissed the violation
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+                value: optimisticReportActions as ReportActions,
+            },
+        ],
+        failureData: [
+            // Rolling back the dismissal of the violation
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`,
+                value: allTransactionViolations,
+            },
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+                value: {
+                    [optimisticReportAction.reportActionID]: null,
```

Comment:
Just want to make sure: which offline pattern do we follow here? Is it B (Optimistic WITH Feedback Pattern)?

Reply:
it's from the other branch that has already been merged, but @yuwenmemon probably has more context regarding this action